### PR TITLE
Feature/swagger

### PR DESCRIPTION
**Pull request by Mauro-Rizzi**

### CODE REVIEW

**Comment by Mauro-Rizzi** [comment]
Will update the documentation on the new api bindings when their pull requests are merged into main.

File: src/api/config/handlers_api.go [hunk 1]
```diff
@@ -12,6 +15,16 @@ import (
...
 )
 
+// @title           PSA Projects API
+// @version         1.0
+// @description     This API gives access to the projects module.
+
+// @license.name  Apache 2.0
+// @license.url   http://www.apache.org/licenses/LICENSE-2.0.html
+
+// @host      localhost:8080
```
Comment (julitaras) [inline src/api/config/handlers_api.go]: Luego deberiamos actualizar esto cuando hagamos el deploy en heroku. O ya poner el dominio de heroku

**Review by julitaras — APPROVED** [review]

**Comment by julitaras** [comment]
Please update with main and fix problems

File: src/api/config/handlers_api.go (hunk 1 above)
Comment (Mauro-Rizzi) [reply src/api/config/handlers_api.go]: @AlexDaciuk Tenes el domain de heroku?

File: src/api/project/api/handler_project_api.go [hunk 2]
```diff
@@ -16,7 +16,17 @@ type ProjectHandler struct {
...
 }
 
-//GetAll handler
+// GetAll ProjectGetter godoc
+// @Summary      Get all projects
+// @Description  Get all the projects in the system
+// @Tags         Projects
+// @Accept       json
+// @Produce      json
+// @Param        task body dto.Project true "Get all projects"
```
Comment (julitaras) [inline src/api/project/api/handler_project_api.go]:
```suggestion
// @Param        project body dto.Project true "Get all projects"
```

File: src/api/project/api/handler_project_api.go [hunk 3]
```diff
@@ -28,7 +38,17 @@ func (dh *ProjectHandler) GetAll(g *gin.Context) {
...
 }
 
-//GetByID handler
+// GetByID ProjectGetterByID godoc
+// @Summary      Get a project
+// @Description  Get a specific project based on it's ID
+// @Tags         Projects
+// @Accept       json
+// @Produce      json
+// @Param        task body dto.Project true "Get a project"
```
Comment (julitaras) [inline src/api/project/api/handler_project_api.go]: idem here.
```suggestion
// @Param        project body dto.Project true "Get a project"
```

File: src/api/project/api/handler_project_api.go [hunk 4]
```diff
@@ -40,7 +60,17 @@ func (dh *ProjectHandler) GetByID(g *gin.Context) {
...
 }
 
-// Post handler
+// Post ProjectCreator godoc
+// @Summary      Add a project
+// @Description  Add a project to the system
+// @Tags         Projects
+// @Accept       json
+// @Produce      json
+// @Param        task body dto.Project true "Create a project"
```
Comment (julitaras) [inline src/api/project/api/handler_project_api.go]:
```suggestion
// @Param        project body dto.Project true "Create a project"
```

File: src/api/project/api/handler_project_api.go [hunk 5]
```diff
@@ -66,6 +96,17 @@ func (ph *ProjectHandler) Post(g *gin.Context) {
 
 }
 
+// Patch ProjectStateUpdater godoc
+// @Summary      Update a project's state
+// @Description  Update a project's state that is already in the system
+// @Tags         Projects
+// @Accept       json
+// @Produce      json
+// @Param        task body dto.Project true "Update a project's state"
```
Comment (julitaras) [inline src/api/project/api/handler_project_api.go]:
```suggestion
// @Param        project body dto.Project true "Update a project's state"
```

File: src/api/project/api/handler_project_api.go [hunk 6]
```diff
@@ -101,6 +142,17 @@ func (ph *ProjectHandler) Patch(g *gin.Context) {
 	g.JSON(http.StatusOK, dto.FromModel(dm))
 }
 
+// Put ProjectPatcher godoc
+// @Summary      Modify a project
+// @Description  Modify a project that is already in the system
+// @Tags         Projects
+// @Accept       json
+// @Produce      json
+// @Param        task body dto.Project true "Create a project"
```
Comment (julitaras) [inline src/api/project/api/handler_project_api.go]:
```suggestion
// @Param        project body dto.Project true "Create a project"
```

**Review by julitaras — CHANGES_REQUESTED** [review]

**Review by julitaras — APPROVED** [review]

**Review by PradaJoaquin — APPROVED** [review]

**Pull request merged**